specs/CookieManagement.md

```diff
@@ -0,0 +1,505 @@
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Get the cookie object at the given index.
+  HRESULT GetValueAtIndex([in] UINT index, [out, retval] ICoreWebView2Cookie** cookie);
+}
+
+/// The caller implements this method to receive the result of the
+/// GetCookies method. The result is written to the cookie list provided in
+/// the GetCookies method call.
+[uuid(0AD3D432-69E9-4223-9CC4-460C20BDCEF5), object, pointer_default(unique)]
+interface ICoreWebView2GetCookiesCompletedHandler : IUnknown {
+  /// Called to provide the implementer with the completion status
+  /// of the corresponding asynchronous method call.
+  HRESULT Invoke(HRESULT result, ICoreWebView2CookieList* cookieList);
+}
+```
+
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    // ...
+
+    /// Kind of cookie SameSite status used in the CoreWebView2Cookie class.
+    /// These fields match those as specified in https://developer.mozilla.org/docs/Web/HTTP/Cookies#.
+    /// Learn more about SameSite cookies here: https://tools.ietf.org/html/draft-west-first-party-cookies-07
+    enum CoreWebView2CookieSameSiteKind
+    {
+        /// None SameSite type. No restrictions on cross-site requests.
+        None = 0,
+        /// Lax SameSite type. The cookie will be sent with "same-site" requests, and with "cross-site" top level navigation.
+        Lax = 1,
+        /// Strict SameSite type. The cookie will only be sent along with "same-site" requests.
+        Strict = 2,
+    };
+
+    runtimeclass CoreWebView2
+    {
+        /// Gets the cookie manager object associated with this CoreWebView2.
+        CoreWebView2CookieManager CookieManager { get; };
+
+        // ...
+    }
+
+    /// Creates, adds or updates, gets, or or view the cookies. The changes would
+    /// apply to the context of the user profile. That is, other WebViews under the
+    /// same user profile could be affected.
+    runtimeclass CoreWebView2CookieManager
+    {
+        /// Create a cookie object with a specified name, value, domain, and path.
+        /// One can set other optional properties after cookie creation.
+        /// This only creates a cookie object and it is not added to the cookie
+        /// manager until you call AddOrUpdateCookie.
+        CoreWebView2Cookie CreateCookieWithDetails(String name, String value, String Domain, String Path);
+
+        /// Creates a cookie whose params matches those of the specified cookie.
+        CoreWebView2Cookie CreateCookie(CoreWebView2 cookie);
+
+        /// Gets a list of cookies matching the specific URI.
+        /// If uri is empty string or null, all cookies under the same profile are
+        /// returned.
+        /// You can modify the cookie objects, call
+        /// CoreWebView2CookieManager.AddOrUpdateCookie, and the changes
+        /// will be applied to the webview.
+        Windows.Foundation.IAsyncOperation<IVectorView> GetCookiesAsync(String uri);

```

Comment:
`IVectorView` of what? CoreWebView2Cookie presumably.

Reply:
Recommend: Yes pls update

Reply:
Upon closer reading, I think it's trying to return a `ICoreWebView2CookieList`. But I would prefer the IVectorView rather than inventing a new interface, assuming it's possible to use IVectorView here. Should `ICoreWebView2CookieList` match `IVectorView`? Or does the `ICoreWebView2CookieList` follow an existing pattern in WebView2 for collections?